File: CsLox/com/craftinginterpreters/lox/LoxLinkedListAdd.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.craftinginterpreters.lox
{
    internal class LoxLinkedListAdd : LoxCallable
    {
        readonly LoxLinkedList list;

        public LoxLinkedListAdd(LoxLinkedList list)
        {
            this.list = list;
        }

        public int arity()
        {
            return 1;
        }

        public Object call(Interpreter interpreter, List<Object> arguments)
        {
            Object value = arguments[0];
            list.elements.Add(value);
            return (list.elements.Count - 1);
        }
    }
}

File: JavaLox/src/com/craftinginterpreters/lox/LoxLinkedListAdd.java
package com.craftinginterpreters.lox;

import java.util.List;

class LoxLinkedListAdd implements LoxCallable {
   final LoxLinkedList list;
   
   public LoxLinkedListAdd(LoxLinkedList list) {
      this.list = list;
   }
   
   @Override
   public int arity() {
      return 1;
   }

   @Override
   public Object call(Interpreter interpreter, List<Object> arguments) {
      Object value = arguments.get(0);
      return list.elements.add(value);
   }
}
